gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveUDFTransformer.scala

```diff
@@ -22,34 +22,37 @@ import org.apache.spark.sql.catalyst.expressions._
 
 import java.util.Locale
 
-object HiveSimpleUDFTransformer {
-  def isHiveSimpleUDF(expr: Expression): Boolean = {
+object HiveUDFTransformer {
+  def isHiveUDF(expr: Expression): Boolean = {
     expr match {
-      case _: HiveSimpleUDF => true
+      case _: HiveSimpleUDF | _: HiveGenericUDF => true
       case _ => false
     }
   }
 
   def replaceWithExpressionTransformer(
       expr: Expression,
       attributeSeq: Seq[Attribute]): ExpressionTransformer = {
-    if (!isHiveSimpleUDF(expr)) {
-      throw new UnsupportedOperationException(s"Expression $expr is not a HiveSimpleUDF")
+    val udfName = expr match {
+      case s: HiveSimpleUDF =>
+        s.name.stripPrefix("default.")
+      case g: HiveGenericUDF =>
+        g.name.stripPrefix("default.")
+      case _ =>
+        throw new UnsupportedOperationException(
```

Comment:
Let's use `GlutenNotSupportException`, which is newly introduced for expected fallback.

Reply:
Thanks for review. Fixed